constraints/src/main/java/gov/nasa/jpl/aerie/constraints/TypescriptCodeGenerationService.java

```diff
@@ -30,12 +30,16 @@ public static String generateTypescriptTypes(
     }
     result.add("}");
 
-    result.add("export type ResourceName = " + String.join(" | ", resources.keySet().stream().map($ -> "\"" + $ + "\"").toList()) + ";");
-    result.add("export type DiscreteResourceSchema<R extends ResourceName> =");
+    result.add("export type Resource = {");
     for (var resource: resources.keySet()) {
-      result.add(indent("R extends \"" + resource + "\" ? " + valueSchemaToTypescript(resources.get(resource)) + " :"));
+      result.add(indent("\"" + resource + "\": " + valueSchemaToTypescript(resources.get(resource)) + ","));
     }
-    result.add(indent("never;"));
+    result.add("}");
+
+    result.add("export type ResourceName = " + String.join(
```

Comment:
I don't remember the answer to this. Did we see an explicit benefit from enumerating them vs using a `keyof Resource` invocation?

Reply:
Yes, now the error message if you mistype the name is 'not assignable to `ResourceName`' instead of 'not assignable to `keyof Resource`', which is less likely to cause confusion